### PR TITLE
fix: hide Check for recording

### LINKED ISSUE #12179: [CAL-2666] regression: dont show recording button in upcoming
also: only show the "Recording" dialog when there is actually a recording. don't show "Check for Recording" for all bookings

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2666](https://linear.app/calcom/issue/CAL-2666/regression-dont-show-recording-button-in-upcoming)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes https://github.com/calcom/cal.com/issues/12179

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -261,13 +261,11 @@ function BookingListItem(booking: BookingItemProps) {
   const title = booking.title;
 
   const showRecordingsButtons = !!(booking.isRecorded && isPast && isConfirmed);
-  const checkForRecordingsButton =
-    !showRecordingsButtons && (booking.location === "integrations:daily" || booking?.location?.trim() === "");
```
Comment (Udit-takkar) [inline apps/web/components/booking/BookingListItem.tsx:265]: completely removing it

**Review by keithwillcode — DISMISSED** [review]
Let’s add an E2E test to verify this doesn’t regress again

**Review by CarinaWolli — APPROVED** [review]

**Comment by Udit-takkar** [comment]
@keithwillcode this won't happen again because I have removed the button completely 😆

**Review by PeerRich — APPROVED** [review]

**Review by PeerRich — DISMISSED** [review]
actually lemme also test on dev

**Comment by PeerRich** [comment]
some checks failing @Udit-takkar

**Comment by Udit-takkar** [comment]
> some checks failing @Udit-takkar

they are not related to this PR. can you retrigger them>

**Pull request merged**